codaco/Server
# Implement auto-updating

Resolves #136.

This adds `electron-updater` to automatically check for & download app updates. I think it makes sense to have this check & notify automatically (following the electron-updater documentation), at least during alpha, but we could tuck this behind a menu item if that's preferable.

### Testing

To test (macOS):
- Downgrade the version number in `src/main/package.json` to `"version": "4.0.0-alpha.1"`
- `npm run build`
- `npm run dist:mac` (requires signing certs)

Expected: Updater sees that alpha.2 has been released, notifies the user, and updates on exit.

<img width="409" alt="notification" src="https://user-images.githubusercontent.com/39674/43606980-dd5b5efa-966a-11e8-917a-bc011f6842aa.png">

### TODO

Test on Windows. I have not been able to create signed builds for Windows, but the package should now be set up to look for the correct certificate.
